src/ntlm/messages/server/authenticate.rs

```diff
@@ -186,8 +186,20 @@ fn process_message_fields(
 
     // will not set workstation because it is not used anywhere
 
-    let mut encrypted_random_session_key = [0x00; ENCRYPTED_RANDOM_SESSION_KEY_SIZE];
-    encrypted_random_session_key.clone_from_slice(message_fields.encrypted_random_session_key.buffer.as_ref());
+    let encrypted_random_session_key: Option<[u8; ENCRYPTED_RANDOM_SESSION_KEY_SIZE]> = match message_fields.encrypted_random_session_key.buffer.len() {
+      0 => None,
+      ENCRYPTED_RANDOM_SESSION_KEY_SIZE => {
+        let mut encrypted_random_session_key = [0x00; ENCRYPTED_RANDOM_SESSION_KEY_SIZE];
+        encrypted_random_session_key.clone_from_slice(message_fields.encrypted_random_session_key.buffer.as_ref());
+        Some(encrypted_random_session_key)
+      },
+      key_len => {
+        return Err(crate::Error::new(
+          crate::ErrorKind::InvalidToken,
+          format!("Encrypted random session key has wrong length. Expected {ENCRYPTED_RANDOM_SESSION_KEY_SIZE} bytes, got {key_len} bytes."),
```

Comment:
Nit: if I’m not mistaken, this `format!` is not required

```suggestion
          "Encrypted random session key has wrong length. Expected {ENCRYPTED_RANDOM_SESSION_KEY_SIZE} bytes, got {key_len} bytes.",
```

Reply:
Hi @CBenoit , thanks for the suggestion!

The ```format!``` is required to get a String (not string-slice) and also to insert the variables ```ENCRYPTED_RANDOM_SESSION_KEY_SIZE``` and ```key_len``` into the error-message.

Removing it, leads to ```expected String, found &str```.
Replacing it with ```String::from()``` leads to an error-message with the variable-names inserted instead of their values.

Reply:
Oh, I’m sorry I missed the format parameter!

Regarding the `expected String, found &str` error, this is probably because you are based on a slightly older version of the codebase. This was changed recently to accept both: https://github.com/Devolutions/sspi-rs/pull/112/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R1776

Anyway, since you are formatting text, you don’t need to change anything, no need to rebase in this case :slightly_smiling_face: